saleor/graphql/account/mutations/permission_group.py

```diff
@@ -215,20 +215,20 @@ class Meta:
         error_type_field = "permission_group_errors"
 
     @classmethod
-    @traced_atomic_transaction()
     def _save_m2m(cls, info, instance, cleaned_data):
         super()._save_m2m(info, instance, cleaned_data)
         remove_users = cleaned_data.get("remove_users")
-        if remove_users:
-            instance.user_set.remove(*remove_users)
-        remove_permissions = cleaned_data.get("remove_permissions")
-        if remove_permissions:
-            instance.permissions.remove(*remove_permissions)
+        with traced_atomic_transaction():
+            if remove_users:
+                instance.user_set.remove(*remove_users)
+            remove_permissions = cleaned_data.get("remove_permissions")
+            if remove_permissions:
+                instance.permissions.remove(*remove_permissions)
```

Comment:
Shouldn't we have whole this method in the transaction? This method is also adding the users, so if something goes wrong during user removal we will end up with saved added users and unsaved removed users.